packages/block/test/block.spec.ts

```diff
@@ -99,10 +100,7 @@ describe('[Block]: block functions', () => {
   })
 
   it('initialization -> setHardfork option', () => {
-    // @ts-ignore type is too strict in this case
-    const common = createCustomCommon(testnetMerge.default, Mainnet, {
-      name: 'testnetMerge',
-    })
+    const common = createCustomCommon(testnetMerge.default as ChainConfig, Mainnet)
```

Comment:
It is not possible to adjust the testdata itself, as TypeScript is not able to infer the literal string types from JSON (it treats all strings as "string", and does not have the ability to recognize that a given string starts with `0x`). The main blocker here is that ChainConfig is expected to contain some `PrefixedHexString` data. An alternative would be to use typescript files and object for testing data instead of json files (at least for these cases). I mentioned the possibility on Discord, but was not comfortable making that choice on my own.

We could however modify ChainConfig to just accept strings rather than PrefixedHexString. We do arguably lose some accuracy by doing that, but it's another alternative to typecasting.

Thoughts?

Reply:
Ok, I just looked this up, but this is *only* about changing/adopting the test data itself anyhow, right? So in `Common` itself, the constructors like `export function createCustomCommon(partialConfig: Partial<ChainConfig>,...` or also simply `constructor(opts: CommonOpts)` already take a TypeScript `ChainConfig` (there is no such thing like "passing JSON" there anyhow I guess?

So this would *only* mean renaming files like `testdata.json` to `testdata.ts` and then have inside:

```ts

const testData = {
  // ...
```

and then adjust the imports?

If so, we should *definitely* do that, for *all* test data files (if you do not want to do all by yourself please open an issue). This is not invasive at all and would solve a lot of problems.

Reply:
Indeed! I think we can do that for all test data. 

I assume there are still some cases where we want to directly import JSON from external sources (not sure, but I assume that's a thing?). Anyhow, have opened an issue just for self-tracking purposes, even though I will be doing this myself within the next couple days. https://github.com/ethereumjs/ethereumjs-monorepo/issues/3637